### PR TITLE
Battery check

### PR DESCRIPTION
**Pull request by CameronRP**
This adds the ability for the raspberry pi to read the battery voltage through the ATtiny for example `i2cget -y 1 0x04 0x20 w` 
Also the ATtiny will now shut down the raspberry pi if the battery voltage gets too low to prevent over discharging the battery (still needs to be fully tested)

### CODE REVIEW

File: attiny/attiny.ino [hunk 1]
```diff
@@ -35,22 +40,72 @@ enum State {
...
   TinyWireS.onReceive(receiveEvent);
   TinyWireS.onRequest(requestEvent);
   setBlinks(2);
-  wdt_enable(WDTO_8S); // enable 8 second WDT
 }
 
-void loop() { 
+void checkBattery() {
+  int a = analogRead(BATTERY_VOLTAGE_PIN);
+  if (425 <= a && a <= 645) {           // Check if battery is between 5.5V and 6.9V
```
Comment (mjs) [inline attiny/attiny.ino]: constants for these values?

File: attiny/attiny.ino [hunk 2]
```diff
@@ -35,22 +40,72 @@ enum State {
...
+    setup_watchdog_interrpt();          // Use WDT for waking up device every 8 seconds.
+    digitalWrite(PI_POWER_PIN, LOW);    // Single long LED flash to indicate low battery
+    digitalWrite(POWER_LED, HIGH);
+    delay(1000);
+    digitalWrite(POWER_LED, LOW);
+    while (425 <= a && a <= 696) {      // Check if the battery is still flat.
```
Comment (mjs) [inline attiny/attiny.ino]: given this logic is repeated, perhaps extract this in a `is_battery_low` function?  That would remove the need for the comments.

File: attiny/attiny.ino (hunk 1 above)
Comment (mjs) [reply attiny/attiny.ino]: please add a comment that below 5.5V could be that the board is being powered by 5V supply

File: attiny/attiny.ino [hunk 3]
```diff
@@ -35,22 +40,72 @@ enum State {
...
+  }
+  digitalWrite(PI_POWER_PIN, HIGH);
+  wdt_enable(WDTO_8S);
+}
+
+void setup_watchdog_interrpt() {
+  WDTCR |= (1 << WDCE) | (1 << WDE);
+  WDTCR = (1 << WDCE) | (1 << WDP3) | (1 << WDP0);
+  WDTCR |= (1 << WDIE);
+}
```
Comment (mjs) [inline attiny/attiny.ino:101]: These lines could do with some docs :)

File: attiny/attiny.ino [hunk 4]
```diff
@@ -275,4 +348,13 @@ ISR(TIMER1_COMPA_vect) {
   }
 }
 
+bool isI2cReadReg(byte val) {
+  for (int i = 0; i < I2C_READ_REG_LEN; i++) {
+    if (i2cReadRegs[i] == val) {
+      return true;
+    }
+  }
```
Comment (mjs) [inline attiny/attiny.ino:361]: This is a bit overkill. If could just be `if (val == I2C_READ_BATTERY_VOLTAGE)  { ...`

**Review by mjs — COMMENTED** [review]
Nice stuff. Just some little things.

File: attiny/attiny.ino (hunk 4 above)
Comment (CameronRP) [reply attiny/attiny.ino:361]: This was done so I can add more registers in the future to read. (see if the WDT was triggered for a restart is one idea)

File: attiny/attiny.ino (hunk 2 above)
Comment (CameronRP) [reply attiny/attiny.ino]: This logic uses a different value for the high value (696 not 645) This is so if the device is being solar charged or small changes in the battery voltage reading won't make it turn on again. 696 is about 7.2V. I'll add a comment for this better explaining it.

File: attiny/attiny.ino (hunk 4 above)
Comment (mjs) [reply attiny/attiny.ino:361]: I figured that but for a small number of registers you could just have an if or switch.

Leaving it like this is fine though.

File: attiny/attiny.ino [hunk 5]
```diff
@@ -69,13 +73,13 @@ void setup() {
 
 void checkBattery() {
   int a = analogRead(BATTERY_VOLTAGE_PIN);
-  if (425 <= a && a <= 645) {           // Check if battery is between 5.5V and 6.9V
+  if (BATTERY_5V5 <= a && a <= BATTERY_6V9) {           // Check if battery is between 5.5V and 6.9V. If the voltage is bellow 5.5V the device will be powered from a 5V wall adapter.
```
Comment (mjs) [inline attiny/attiny.ino:76]: below

File: attiny/attiny.ino [hunk 6]
```diff
@@ -90,9 +94,10 @@ void checkBattery() {
 }
 
 void setup_watchdog_interrpt() {
-  WDTCR |= (1 << WDCE) | (1 << WDE);
-  WDTCR = (1 << WDCE) | (1 << WDP3) | (1 << WDP0);
-  WDTCR |= (1 << WDIE);
+  // For more detail see section 8.4 of the ATtiny 85 datasheet http://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-2586-AVR-8-bit-Microcontroller-ATtiny25-ATtiny45-ATtiny85_Datasheet.pdf
+  WDTCR |= (1 << WDCE) | (1 << WDE);  // These bits have to be set before disabling the WDT (have to disable the WDT restart to enable the WDT interrupt)
+  WDTCR = (1 << WDCE) | (1 << WDP3) | (1 << WDP0);  // Disable the WDT and set the interval to 8 seconds
+  WDTCR |= (1 << WDIE); // This will enable the WDT interrupt (not WDT restart).
```
Comment (mjs) [inline attiny/attiny.ino:100]: Nice. This is much better.

File: attiny/attiny.ino [hunk 7]
```diff
@@ -25,6 +25,10 @@
 #define MINUTE_COUNTDOWN 108 // = (60/0.5)*90% as each interupt occurs every 0.5 seconds. -10% because of inaccurate internal clock
 #define PI_WDT_RESET_VAL 30000 // 5 minutes 100*60*5
 #define BATTERY_VOLTAGE_PIN A2
+#define BATTERY_5V5 425
+#define BATTERY_6V9 645
+#define BATTERY_7V2 696
```
Comment (mjs) [inline attiny/attiny.ino:30]: great

**Review by mjs — APPROVED** [review]
Excellent. LGTM.

**Pull request merged**